clcache.py

```diff
@@ -1526,18 +1527,18 @@ def processCompileRequest(cache, compiler, args):
     printTraceStatement("Expanded commandline '{0!s}'".format(cmdLine))
 
     try:
-        sourceFiles, objectFile = CommandLineAnalyzer.analyze(cmdLine)
+        sourceFiles, objectFiles = CommandLineAnalyzer.analyze(cmdLine)
 
         if len(sourceFiles) > 1:
             return reinvokePerSourceFile(cmdLine, sourceFiles, environment), '', ''
         else:
             assert objectFile is not None
             if 'CLCACHE_NODIRECT' in os.environ:
                 returnCode, compilerOutput, compilerStderr, cleanupRequired = \
-                    processNoDirect(cache, objectFile, compiler, cmdLine, environment)
+                    processNoDirect(cache, objectFiles[0], compiler, cmdLine, environment)
             else:
                 returnCode, compilerOutput, compilerStderr, cleanupRequired = \
-                    processDirect(cache, objectFile, compiler, cmdLine, sourceFiles[0])
+                    processDirect(cache, objectFiles[0], compiler, cmdLine, sourceFiles[0])
```

Comment:
What is the benefit of doing this patch - it appears that the code always only uses the first list element? If this is a preparation for a subsequent patch, please explain how this helps implementing the next commit. As it is, the commit by itself does not seem very useful.